src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -75,46 +68,21 @@ export default class NativeAdapter implements IMediaSourceAdapter {
    * @static
    */
   static canPlayType(mimeType: string): boolean {
-    let canPlayType = !!(document.createElement("video").canPlayType(mimeType));
+    let canPlayType = (typeof mimeType === 'string') ?
+      !!(document.createElement("video").canPlayType(mimeType.toLowerCase())) : false;
     NativeAdapter._logger.debug('canPlayType result for mimeType:' + mimeType + ' is ' + canPlayType.toString());
     return canPlayType;
   }
 
   /**
-   * Checks if the media source adapter is supported
-   * @function isSupported
-   * @returns {boolean} - Whether the media source adapter is supported. Default implementation is true
-   * @static
-   */
-  static isSupported(): boolean {
-    NativeAdapter._logger.debug('isSupported:true');
-    return true;
-  }
-
-  /**
-   * Factory method to create media source adapter
-   * @function createAdapter
-   * @param {HTMLVideoElement} videoElement - The video element that the media source adapter work with
-   * @param {Object} source - The source Object
-   * @param {Object} config - The media source adapter configuration
-   * @returns {IMediaSourceAdapter} - New instance of the run time media source adapter
-   * @static
-   */
-  static createAdapter(videoElement: HTMLVideoElement, source: Source, config: Object): IMediaSourceAdapter {
-    NativeAdapter._logger.debug('Creating adapter');
-    return new this(videoElement, source, config);
-  }
-
-  /**
    * @constructor
    * @param {HTMLVideoElement} videoElement - The video element which bind to NativeAdapter
    * @param {Source} source - The source object
    * @param {Object} config - The media source adapter configuration
    */
   constructor(videoElement: HTMLVideoElement, source: Source, config: Object) {
-    this._config = config;
-    this._videoElement = videoElement;
-    this._sourceObj = source;
+    NativeAdapter._logger.debug('Creating adapter');
+    super(videoElement, source, config);
     this._eventManager = new EventManager();
```

Comment:
What should be in the base? please explain.
if you are talking about the event manager, so no, hls and dash wont need him.